s3same/__init__.py

```diff
@@ -0,0 +1,39 @@
+import boto3
+import travispy
+import yaml
+
+from .travis import travis_encrypt as _travis_encrypt
+from .iam import credentials_for_new_user, IAMName
+
+class S3same (object):
+    def __init__(self, 
+            repo, pro, github_token, github_owner, s3_bucket,
+            aws_region=None, aws_key=None, aws_secret=None, aws_profile=None):
+        uri = travispy.travispy.PRIVATE if pro else travispy.travispy.PUBLIC
+        self._travis = travispy.TravisPy.github_auth(github_token, uri=uri)
+        self._aws = boto3.session.Session(
+                aws_access_key_id=aws_key,
+                aws_secret_access_key=aws_secret,
+                region_name=aws_region,
+                profile_name=aws_profile)
+        self._repo_slug = '{}/{}'.format(github_owner, repo)
+        self._iam_user = '{}__{}__{}'.format(IAMName, github_owner, repo)
+        self._s3_bucket = s3_bucket
+
+    def artifact_yaml(self):
```

Comment:
This is somewhat my opinion, but unless you plan to expand this class, I think you could just drop it completely and move all the init stuff into `artifact_yaml()` (with maybe a helper function to set up the boto session/IAM client). As it is, it seems like all the class' effects are largely accomplished by methods outside the class anyway, and it's sort of Pythonic to eschew the class and just use the file architecture to handle variable scoping.